common/fs/src/cache/dir_path.rs

```diff
@@ -1,9 +1,17 @@
+use regex::Regex;
 use std::ops::Deref;
 use std::path::{Path, PathBuf};
 use thiserror::Error;
-use tracing::error;
+use tracing::{error, warn};
+
 #[cfg(not(windows))]
-use tracing::warn;
+lazy_static! {
+    static ref REGEX_FS_ROOT: Regex = Regex::new(r"^/$").unwrap();
+}
+#[cfg(windows)]
+lazy_static! {
+    static ref REGEX_FS_ROOT: Regex = Regex::new(r"^ [a-zA-Z]:[\\/]$").unwrap();
```

Comment:
Just curious, would a root path for Windows ever contain a lowercase char?

Reply:
yes